src/preact-render-spy.js

```diff
@@ -191,27 +225,40 @@ class FindWrapper {
   }
 
   at(index) {
+    if (index >= this.length) {
+      throw new Error(`Not enough results to return result ${index}.`);
+    }
+
     return new FindWrapper(this.spy, [this[index]]);
   }
 
   attr(name) {
-    for (const item of Array.from(this)) {
-      if (
-        typeof item === 'object' &&
-        item.attributes &&
-        item.attributes[name]
-      ) {
-        return item.attributes[name];
-      }
+    if (this.length > 1) {
+      throw new Error('Too many results. Can only return attribute for 1 result.');
```

Comment:
I feel like since jest's error messages suck, we should try to make these as informative as possible:

```js

if (this.length !== 1) {
  throw new Error(`preact-render-spy: Must have exactly 1 result for .attr('${name}') - ${this.length} results found`);
}